### PR TITLE
Update files to make it multilanguage

### PR DESCRIPTION
**Pull request by bovirus**
Update NSIS script
Update language files

Now the installer seems full translated.

Please test.

I cannot find in the script the are **_i386 file section_** and **_alpha file section_**.

I cannot test the icon description because the installer didn't create program group.

Thanks.

### CODE REVIEW

File: installer/installer_strings_english.nsh [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
-LangString User_Documentation_Link_Text 1033 "User Documentation"
-LangString Start_Menu_Shortcuts_Section_Text 1033 "Start Menu Shortcuts"
-LangString Uninstall_Link_Text 1033 "Uninstall"
-LangString Uninstall_Section_Text 1033 "Uninstall"
-LangString Technical_Documentation_Link_Text 1033 "Technical Documentation"
-LangString GNOME_desktop_icon_theme_Section_Text 1033 "GNOME desktop icon theme"
-LangString Alpha_system_emulation_Section_Text 1033 "Alpha system emulation"
-LangString PC_i386_system_emulation_Section_Text 1033 "PC i386 system emulation"
-LangString Menu_entries_Section_Text 1033 "Menu entries"
+;-------------------------------- Installer text strings (English - LangID ${LANG_English}3)
```
Comment (stweil) [inline installer/installer_strings_english.nsh:1]: Why does the comment include this: `${LANG_English}3`?

File: installer/installer_strings_italian.nsh [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
-LangString User_Documentation_Link_Text 1040 "Documentazione utente (Inglese)"
-LangString Start_Menu_Shortcuts_Section_Text 1040 "Collegamenti nel menu Start"
-LangString Uninstall_Link_Text 1040 "Disinstalla"
-LangString Uninstall_Section_Text 1040 "Disinstalla"
-LangString Technical_Documentation_Link_Text 1040 "Documentazione tecnica"
-LangString GNOME_desktop_icon_theme_Section_Text 1040 "Tema icon edesktop Gnome"
-LangString Alpha_system_emulation_Section_Text 1040 "Emulazione sistema alpha"
-LangString PC_i386_system_emulation_Section_Text 1040 "Emulazione sistema PC i386"
-LangString Menu_entries_Section_Text 1040 "Voci menu"
+;-------------------------------- Installer text strings (Italian - LangID ${LANG_Italian})
```
Comment (stweil) [inline installer/installer_strings_italian.nsh:1]: See comment above.

File: qemu.nsi [hunk 3]
```diff
@@ -239,27 +237,33 @@ Section "${Uninstall_Section_Text}"
...
 
-;--------------------------------
+; Include files with language instalelr strings.
+; Language ID table - https://www.science.co.il/language/Locale-codes.php
+; Language ID 1033 - English
+; Language ID 1031 - German
+; Language ID 1034 - Spanish
+; Language ID 1036 - French
+; Language ID 1040 - Italian
+!addincludedir installer
```
Comment (stweil) [inline qemu.nsi:247]:
```suggestion
!addincludedir ${SRCDIR}/installer
```

File: qemu.nsi [hunk 4]
```diff
@@ -239,27 +237,33 @@ Section "${Uninstall_Section_Text}"
...
 !endif
 !ifdef CONFIG_DOCUMENTATION
-    !insertmacro MUI_DESCRIPTION_TEXT ${SectionDoc}     "${Documentation_Section_Text}"
+    !insertmacro MUI_DESCRIPTION_TEXT ${Documentation_Section_Description}     $(Documentation_Section_Description_Text)
 !endif
-    !insertmacro MUI_DESCRIPTION_TEXT ${SectionMenu}    "${Menu_entries_Section_Text}"
+    !insertmacro MUI_DESCRIPTION_TEXT ${Start_Menu_Section_Description}        $(Start_Menu_Section_Description_Text)
 !insertmacro MUI_FUNCTION_DESCRIPTION_END
 
-;--------------------------------
```
Comment (stweil) [inline qemu.nsi:260]: Why was this code part moved up?

**Review by stweil — CHANGES_REQUESTED** [review]
Thank you for this update. It works basically. Here are some small comments. I also had to add files for German, French and Spanish, because otherwise those languages would not work.

File: installer/installer_strings_english.nsh [hunk 5]
```diff
@@ -1,19 +1,21 @@
...
+LangString Documentation_Section_Name ${LANG_ENGLISH} "Documentation"
+LangString Documentation_Section_Description_Text ${LANG_ENGLISH} "User and technical documentation"
+LangString Start_Menu_Section_Name ${LANG_ENGLISH} "Program Group"
+LangString Start_Menu_Section_Description_Text ${LANG_ENGLISH} "Create program group in Start menu"
+LangString Uninstall_Section_Name ${LANG_ENGLISH} "Uninstall"
+LangString Uninstall_Section_Description_Text ${LANG_ENGLISH} "Create program link to uninstall program"
+LangString Link_Description_User_Documentation ${LANG_ENGLISH} "User documentation"
+LangString Link_Description_Uninstall ${LANG_ENGLISH} "Uninstall"
+LangString Link_Description_Technical_Documentation ${LANG_ENGLISH} "Technical Documentation"
+LangString Visit_QEMU_WIki_Link ${LANG_ENGLISH} "Visit the QEMU Wiki online!"
```
Comment (stweil) [inline installer/installer_strings_english.nsh:21]:
```suggestion
LangString Visit_QEMU_Wiki_Link ${LANG_ENGLISH} "Visit the QEMU Wiki online!"
```

File: installer/installer_strings_italian.nsh [hunk 6]
```diff
@@ -1,19 +1,21 @@
...
+LangString Documentation_Section_Name ${LANG_ITALIAN} "Documentazione"
+LangString Documentation_Section_Description_Text ${LANG_ITALIAN} "Documentazione utente e tecnica"
+LangString Start_Menu_Section_Name ${LANG_ITALIAN} "Gruppo programmi"
+LangString Start_Menu_Section_Description_Text ${LANG_ITALIAN} "Crea gruppo programma in menu Start"
+LangString Uninstall_Section_Name ${LANG_ITALIAN} "Disinstallazione"
+LangString Uninstall_Section_Description_Text ${LANG_ITALIAN} "Crea collegamento per disinstallazione programma"
+LangString Link_Description_User_Documentation ${LANG_ITALIAN} "Documentazione utente"
+LangString Link_Description_Uninstall ${LANG_ITALIAN} "Disinstalla"
+LangString Link_Description_Technical_Documentation ${LANG_ITALIAN} "Documentazione tecnica"
+LangString Visit_QEMU_WIki_Link ${LANG_ITALIAN} "Visita il Wiki QEMU su internet"
```
Comment (stweil) [inline installer/installer_strings_italian.nsh:21]:
```suggestion
LangString Visit_QEMU_Wiki_Link ${LANG_ITALIAN} "Visita il Wiki QEMU su internet"
```

File: qemu.nsi [hunk 7]
```diff
@@ -89,7 +89,7 @@ RequestExecutionLevel admin
 !insertmacro MUI_PAGE_COMPONENTS
 !insertmacro MUI_PAGE_DIRECTORY
 !insertmacro MUI_PAGE_INSTFILES
-!define MUI_FINISHPAGE_LINK "Visit the QEMU Wiki online!"
+!define MUI_FINISHPAGE_LINK $(Visit_QEMU_WIki_Link)
```
Comment (stweil) [inline qemu.nsi:92]:
```suggestion
!define MUI_FINISHPAGE_LINK $(Visit_QEMU_Wiki_Link)
```

**Review by stweil — CHANGES_REQUESTED** [review]

**Comment by stweil** [comment]
@bovirus, I still need a signed commit from you, otherwise your contribution won't be accepted by upstream QEMU (see https://wiki.qemu.org/Contribute/SubmitAPatch).

**Comment by bovirus** [comment]
@stweil 

Could be explain me in details how can help you and what I should do?

**Comment by stweil** [comment]
Your commit must have a commit message which ends with a `Signed-off-by` line like this one:

    Signed-off-by: YOUR_FULL_NAME <EMAIL_NAME@EMAIL_DOMAIN>

(replace the two parts with your full name (first name and family name) and your e-mail address using valid data)

By the way: the latest installers at https://qemu.weilnetz.de/ include your translation. I still need your `Signed-off-by` to integrate your contribution into the official QEMU code. See https://wiki.qemu.org/Contribute/SubmitAPatch#Patch_emails_must_include_a_Signed-off-by:_line.

**Comment by bovirus** [comment]
Hello.

Thanks for the explanation.

Could you please add them as

Signed-off-by: bovirus bovirus@gmail.com <:bovirus@gmail.com>

**Comment by stweil** [comment]
That would not be accepted by the QEMU maintainers. Quote from the contributor guidelines: 

> Please use your real name to sign a patch (not an alias or acronym).

**Comment by bovirus** [comment]
@stweil 

Thanks again for the detail.
Please use

    Signed-off-by: Roberto Boriotti <bovirus@gmail.com>

**Comment by stweil** [comment]
Multilanguage was added meanwhile, so I think this pull request can be closed. Please tell me if something is still missing.

**Comment by bovirus** [comment]
@stweil 
Yes you can close this request because instalelr now is multilanguage.
Thanks.

**Pull request closed without merging**